### PR TITLE
Fix import.meta.url inside web workers

### PR DESCRIPTION
**Pull request by ceifa**
This PR contains:
- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [x] yes
- [ ] no

Breaking Changes?
- [ ] yes
- [x] no

### Description

If I compile a library that uses `import.meta.url` internally using UMD, it will break if I try to use it inside web workers, that happens because today rollup compile `import.meta.url` to use nodejs require or browser document API, but both is unavailable inside web workers. This PR adds a check for it and try to use `location.href`, if available.

### CODE REVIEW

**Comment by guybedford** [comment]
Thanks for the PR.

How about we flip the order and make it `if (typeof document === 'undefined' ? typeof location !== 'undefined' ? location.href : ...`?

**Comment by ceifa** [comment]
> How about we flip the order and make it `if (typeof document === 'undefined' ? typeof location !== 'undefined' ? location.href : ...`?

Thanks for reviewing! Just pushed this suggestion. :)

File: test/form/samples/emit-uint8array-no-buffer/_expected/umd.js [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
 
-	var asset2b = (typeof document === 'undefined' ? new (require('u' + 'rl').URL)('file:' + __dirname + '/assets/asset-52cbd095').href : new URL('assets/asset-52cbd095', document.currentScript && document.currentScript.src || document.baseURI).href);
+	var asset2b = (typeof document === 'undefined' ? (typeof location !== 'undefined' ? location.href : new (require('u' + 'rl').URL)('file:' + __dirname + '/assets/asset-52cbd095').href) : new URL('assets/asset-52cbd095', document.currentScript && document.currentScript.src || document.baseURI).href);
 
-	var asset99a = (typeof document === 'undefined' ? new (require('u' + 'rl').URL)('file:' + __dirname + '/assets/asset-c568a840').href : new URL('assets/asset-c568a840', document.currentScript && document.currentScript.src || document.baseURI).href);
+	var asset99a = (typeof document === 'undefined' ? (typeof location !== 'undefined' ? location.href : new (require('u' + 'rl').URL)('file:' + __dirname + '/assets/asset-c568a840').href) : new URL('assets/asset-c568a840', document.currentScript && document.currentScript.src || document.baseURI).href);
```
Comment (guybedford) [inline test/form/samples/emit-uint8array-no-buffer/_expected/umd.js:14]: Apologies I didn't review this fully - we need to ensure we are doing the correct normalization here:

```suggestion
	var asset99a = (typeof document === 'undefined' && typeof location === 'undefined' ? new (require('u' + 'rl').URL)('file:' + __dirname + '/assets/asset-c568a840').href) : new URL('assets/asset-c568a840', typeof document !== 'undefined' ? document.currentScript && document.currentScript.src || document.baseURI : location.href).href);
```

Comment (ceifa) [reply test/form/samples/emit-uint8array-no-buffer/_expected/umd.js]: Fixed

**Review by guybedford — APPROVED** [review]

**Pull request merged**